soci/soci_index.go

```diff
@@ -305,18 +305,42 @@ func NewIndexBuilder(contentStore content.Store, blobStore orascontent.Storage,
 	}, nil
 }
 
+// BuildAndPush builds a SOCI index, pushes and labels the artifacts, and returns the SOCI index.
+func (b *IndexBuilder) BuildAndPush(ctx context.Context, blobStore store.Store, img images.Image) (*IndexWithMetadata, error) {
```

Comment:
It is a little odd to me the Builder interface is changing to accommodate a behavior specific to one of the blob store implementations. Can the handling of the lease be pushed into the containerd blob store?

Reply:
Honestly the entire building interface is super weird IMO. Why are we building an index and pushing zTOCs before labeling them? The flow naturally leads to race conditions and so this was the best way I could think of without more refactoring work. But, I could put in more time to refactor it deeper, it just seemed unnecessary.

> Can the handling of the lease be pushed into the containerd blob store?

IMO it shouldn't since, while SOCI content store does not garbage collect, we could theoretically support it in the future, and in fact it might be something we want to pursue, since leftover artifacts not being cleaned up on some operations (e.g. image removal with a SOCI image only removes image manifests) can lead to unexpected behaviors.